features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/async/RecordedDataQueueHandler.kt

```diff
@@ -170,8 +124,14 @@ internal class RecordedDataQueueHandler : DataQueueHandler {
      */
     @WorkerThread
     @Synchronized
-    private fun triggerProcessingLoop(currentTime: Long) {
+    private fun triggerProcessingLoop(initialTime: Long) {
+        var currentTime = initialTime
+
         while (recordedDataQueue.isNotEmpty()) {
+            mainThreadHandler.post {
+                currentTime = timeProvider.getDeviceTimestamp()
+            }
+
```

Comment:
I'm not sure it is a right solution. Maybe it can reduce the problem somehow, but the issue here is concurrency: `mainThreadHandler.post` is non-blocking call, so by the time `currentTime = timeProvider.getDeviceTimestamp()` is executed, you can be already advance in the `while` loop past `post` line, or even be on the next turn.

Reply:
I've changed the implementation to have a separate property to store `System.nanoTime` for each item and query that, so it will be monotonic and we're no longer dependent upon calling on a certain thread